src/test/java/com/airlines/dao/impl/CrewDaoImplTest.java

```diff
@@ -0,0 +1,182 @@
+package com.airlines.dao.impl;
+
+import com.airlines.dao.CrewDao;
+import com.airlines.model.Crew;
+import com.airlines.model.CrewMember;
+import com.airlines.util.JdbcUtil;
+import com.airlines.util.TestDatasourceProvider;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import javax.sql.DataSource;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.airlines.model.Citizenship.BRAZILIAN;
+import static com.airlines.model.Citizenship.ITALIAN;
+import static com.airlines.model.Position.PILOT_IN_COMMAND;
+import static org.junit.Assert.*;
+
+public class CrewDaoImplTest {
+    private static final String SCHEMA_SQL_PATH = "src/test/resources/sql/schema.sql";
+    private static final String DATA_SQL_PATH = "src/test/resources/sql/test_data.sql";
+    private static final String CREATE_TABLES_QUERY = JdbcUtil.getSqlQueryString(SCHEMA_SQL_PATH);
+    private static final String GENERATE_DATA_QUERY = JdbcUtil.getSqlQueryString(DATA_SQL_PATH);
+
+    private static final CrewMember CREW_MEMBER = buildTestCrewMember();
+    private static final Crew CREW = buildCrewFromDataSet();
+    private static final Crew INVALID_CREW = buildInvalidCrewFromDataSet();
+
+    private static CrewDao crewDao;
+
+    private static DataSource dataSource;
+
+    @BeforeClass
+    public static void init() {
+        dataSource = TestDatasourceProvider.createTestDatasource();
+        crewDao = new CrewDaoImpl(dataSource);
+    }
+
+    @Before
+    public void dataBaseInit() throws SQLException {
+        createTables();
+    }
+
+    @Test
+    public void shouldThrowExceptionIfInvalidArgumentCrewMemberProvidedToAddMethod() {
+        IllegalArgumentException exception = assertThrows(IllegalArgumentException.class, () -> crewDao.add(null, null));
+        assertEquals("Valid crewMember entity should be provided", exception.getMessage());
+    }
+
+    @Test
+    public void shouldThrowExceptionIfInvalidArgumentCrewProvidedToAddMethod() {
```

Comment:
crewDao.add contains code:
 ```
    if (crewMember == null || crewMember.getId() == null) {
            throw new IllegalArgumentException("Valid crewMember entity should be provided");
        }
        if (crew == null || crew.getId() == null) {
            throw new IllegalArgumentException("Valid crew entity should be provided");
        }
```
Usually, each IF condition it's 1 test, we have there 4 conditions (2 conditions for the first IF and 2 for the second).
Maybe we could even cover it with parametrized test, but in junit 4 it could require some effort, so separate test good enough:

```
    @Test
    public void shouldThrowExceptionIfCrewMemberWithNoIdProvidedToAddMethod() {
        CrewMember crewMember = CrewMember.builder()
                .withId(null)
                .build();

        IllegalArgumentException exception = assertThrows(IllegalArgumentException.class, () -> crewDao.add(crewMember, null));

        assertEquals("Valid crewMember entity should be provided", exception.getMessage());
    }
```
and for 2nd IF:
```
    @Test
    public void shouldThrowExceptionIfCrewProvidedWithoutIdToAddMethod() {
        Crew crew = new Crew(null, "Some name", Collections.singletonList(CREW_MEMBER));

        IllegalArgumentException exception = assertThrows(IllegalArgumentException.class, () -> crewDao.add(CREW_MEMBER, crew));

        assertEquals("Valid crew entity should be provided", exception.getMessage());
    }
```